### PR TITLE
Add 106-Ounce Cans

### PR DESCRIPTION
**Pull request by Kanexan**
#### Summary
```SUMMARY: Content "Adds 106-ounce (3L) food cans."```

#### Purpose of change
This is to add economy/bulk-size aluminum cans of food, such as one would use for commercial food prep or get from a warehouse store, and reduce the weight of empty standard-size cans to 40 from 400.

#### Describe the solution
This adds the cans into the game, and places them into the item groups for kitchens and general canned goods at roughly the same frequency as the similarly-sized 3-liter glass jars. The reduced-weight thing for small cans is because they weigh ten times more in-game than they do in reality.

#### Additional context
image
A proof-of-concept on cans IRL weighing about 40 grams.

### CODE REVIEW

**Comment by Mecares** [comment]
I guess you mean 40 grams and not ounces?

**Comment by Kanexan** [comment]
> 
> 
> I guess you mean 40 grams and not ounces?

I did, thank you for pointing that out.

**Comment by roaringjohn** [comment]
You need to lint.

**Comment by anothersimulacrum** [comment]
Jenkins rebuild

File: data/json/item_groups.json [hunk 1]
```diff
@@ -627,7 +627,12 @@
       { "item": "veggy_pickled", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
       { "item": "fish_pickled", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
       { "item": "sauce_red", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
-      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" }
+      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
+      { "item": "can_beans", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_corn", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_tomato", "prob": 4, "charges": 12, "container-item": "can_food_big" },
```
Comment (anothersimulacrum) [inline data/json/item_groups.json]:
```suggestion
      { "item": "can_tomato", "prob": 4, "charges": 24, "container-item": "can_food_big" },
```

File: data/json/item_groups.json [hunk 2]
```diff
@@ -627,7 +627,12 @@
       { "item": "veggy_pickled", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
       { "item": "fish_pickled", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
       { "item": "sauce_red", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
-      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" }
+      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
+      { "item": "can_beans", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_corn", "prob": 4, "charges": 12, "container-item": "can_food_big" },
```
Comment (anothersimulacrum) [inline data/json/item_groups.json]:
```suggestion
      { "item": "can_corn", "prob": 4, "charges": 4, "container-item": "can_food_big" },
```

File: data/json/item_groups.json [hunk 3]
```diff
@@ -627,7 +627,12 @@
...
+      { "item": "can_tomato", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_pineapple", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "sauce_red", "prob": 3, "charges": 12, "container-item": "can_food_big" }
```
Comment (anothersimulacrum) [inline data/json/item_groups.json]:
```suggestion
      { "item": "sauce_red", "prob": 3, "charges": 96, "container-item": "can_food_big" }
```

File: data/json/item_groups.json [hunk 4]
```diff
@@ -1855,7 +1860,12 @@
       { "item": "veggy_pickled", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
       { "item": "fish_pickled", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
       { "item": "sauce_red", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
-      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" }
+      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
+      { "item": "can_beans", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_tomato", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_pineapple", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_corn", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "sauce_red", "prob": 3, "charges": 12, "container-item": "can_food_big" }
```
Comment (anothersimulacrum) [inline data/json/item_groups.json]:
```suggestion
      { "item": "sauce_red", "prob": 3, "charges": 96, "container-item": "can_food_big" }
```

File: data/json/item_groups.json [hunk 5]
```diff
@@ -1855,7 +1860,12 @@
       { "item": "veggy_pickled", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
       { "item": "fish_pickled", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
       { "item": "sauce_red", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
-      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" }
+      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
+      { "item": "can_beans", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_tomato", "prob": 4, "charges": 12, "container-item": "can_food_big" },
```
Comment (anothersimulacrum) [inline data/json/item_groups.json]:
```suggestion
      { "item": "can_tomato", "prob": 4, "charges": 24, "container-item": "can_food_big" },
```

File: data/json/item_groups.json [hunk 6]
```diff
@@ -1855,7 +1860,12 @@
...
+      { "item": "can_pineapple", "prob": 4, "charges": 12, "container-item": "can_food_big" },
+      { "item": "can_corn", "prob": 4, "charges": 12, "container-item": "can_food_big" },
```
Comment (anothersimulacrum) [inline data/json/item_groups.json]:
```suggestion
      { "item": "can_corn", "prob": 4, "charges": 4, "container-item": "can_food_big" },
```

**Comment by Kanexan** [comment]
I get how to make them their own itemgroup, but if I make them an itemgroup of their own, how do I then get them to spawn in kitchens, or where canned goods would ordinarily spawn?

**Comment by anothersimulacrum** [comment]
You nest that itemgroup inside the itemgroups that you would put them in otherwise.
So instead of the `cannedfood` itemgroup having
```json
      { "item": "kompot", "prob": 4, "charges": 12, "container-item": "jar_3l_glass_sealed" },
      { "item": "can_beans", "prob": 4, "charges": 12, "container-item": "can_food_big" },
      { "item": "can_corn", "prob": 4, "charges": 4, "container-item": "can_food_big" },
      { "item": "can_tomato", "prob": 4, "charges": 24, "container-item": "can_food_big" },
      { "item": "can_pineapple", "prob": 4, "charges": 12, "container-item": "can_food_big" },
      { "item": "sauce_red", "prob": 3, "charges": 96, "container-item": "can_food_big" }
```
it has
```json
      { "group": "big_canned_food", "prob": 4 }
```
and `big_canned_food` has that.

**Comment by Rivet-the-Zombie** [comment]
I've been waiting for some time for somebody to PR us some #10 cans. Just imagine finding an unopened flat of a dozen of these; you'd be set for canned stuff for a week or three!

**Comment by Kanexan** [comment]
I'm just kinda surprised nobody caught the thing about the can weight, honestly. Glad you like the idea!
And also thanks a bunch, Simulacrum, you've been extremely helpful on this and I'm glad you took the time to point out some of my mistakes.

**Pull request merged**